src/libvirt-dbus.js

```diff
@@ -369,18 +369,16 @@ const LIBVIRT_DBUS_PROVIDER = {
                 }
             }
 
-            // FIXME: use Promise.all() here; but that causes "Error: Actions must be plain objects"
-            // eslint-disable-next-line cockpit/no-cockpit-all
-            return cockpit.all(storageVolPromises)
+            return Promise.all(storageVolPromises)
                     .then(() => {
-                        return call(connectionName, objPath, 'org.libvirt.Domain', 'Undefine', [flags], { timeout, type: 'u' });
+                        return dispatch(call(connectionName, objPath, 'org.libvirt.Domain', 'Undefine', [flags], { timeout, type: 'u' }));
```

Comment:
I am actually not 100% sure if the dispatch() needs to go here or somewhere else -- I'm a redux n00b. Hence, @KKoukiou , I'd appreciate extra scrutiny here. Thanks!